cfme/tests/infrastructure/test_provisioning_dialog.py

```diff
@@ -115,8 +141,33 @@ def _provisioner(template, provisioning_data, delayed=None):
     return _provisioner
 
 
+def check_all_tabs(provision_request, provider):
+    view = navigate_to(provision_request, "Details")
+    widgets_names = 'request', 'purpose', 'catalog', 'environment', 'schedule'
+    if provider.one_of(EC2Provider):
+        widgets_names += 'properties', 'customize'
+    elif provider.one_of(CloudProvider):
+        widgets_names += 'properties', 'volumes', 'customize'
+    elif provider.one_of(SCVMMProvider):
+        widgets_names += 'environment', 'hardware', 'network'
+    elif provider.one_of(InfraProvider):
+        widgets_names += 'hardware', 'network', 'customize'
+    else:
```

Comment:
_Optional_ I am not crazy about `if` blocks once they get over 3 conditions, I think this is the kind of thing that would be well suited for a dict or a small yaml file.

Reply:
The order of checking the providers is important. I noted this in a code.

I think that ideally the branching should not be there at all and the expected widget names should be defined in provider class hierarchy instead. Something like
```
class BaseProvider:
    expected_widgets_names = 'request', 'purpose', 'catalog', 'environment', 'schedule'
    ...

class CloudProvider(BaseProvider):
    expected_widgets_names = BaseProvider.expected_widgets_names + ['properties', 'volumes', 'customize']
    ...

class EC2Provider(CloudProvider):
    expected_widgets_names = BaseProvider.expected_widgets_names + ['properties', 'customize']
    ...
```

What do you think about this?

Reply:
I think it'd be better to have it be a part of provider `data`, since these are hardcoded strings. I think we can leave it as the `if/elif/else` block for this PR though and you can make changes to it later.

Reply:
> I think it'd be better to have it be a part of provider `data`, since these are hardcoded strings. I think we can leave it as the `if/elif/else` block for this PR though and you can make changes to it later.

I am strongly against keeping this in data. We have many of strings of similar kind hardcoded. Do `git grep title=` -- you will get 400 occurrences. And I think this is just a tip of the iceberg. I think it solves nothing.

I see `if`/`elif`/`else` together with `instanceof` on many places in the code and I think this is a code smell. Perhaps quick but dirty solution to a problem. I think it is nicely explained here https://stackoverflow.com/a/30894403/1950100